backends/dpdk/dpdkContext.cpp

```diff
@@ -451,6 +466,199 @@ void DpdkContextGenerator::addMatchTables(Util::JsonArray* tablesJson) {
+    if (resnameexpr->is<IR::StringLiteral>() == false) return;
+
+    auto resname = resnameexpr->to<IR::StringLiteral>()->value;
+
+    /* validate resource name */
+    if (resname != "mirror_profile" && resname != "lpm_profile") {
+        ::error("Invalid argument '%1%' for 'intel_lut_type' annotation for '%2%'"
+                " it should be '%3%' or '%4%'.",
+                resname, d->externalName(), "mirror_profile", "lpm_profile");
+        return;
+    }
+
+    auto type = typemap->getType(d);
+    if (type->is<IR::Type_SpecializedCanonical>() == false) return;
+
+    auto exactMVLut = new  P4MatchLookupTableInfo();
+    exactMVLut->handle = getNewMVLutHandle();
+    exactMVLut->ctrlName = ctrl_name;
+    exactMVLut->tblName = d->externalName();
+    exactMVLut->p4Hidden = false;
+    exactMVLut->isP4Hidden = true;
+    if (d->arguments->at(0)->expression->is<IR::Constant>()) {
+        exactMVLut->size = d->arguments->at(0)->expression->to<IR::Constant>()->asInt();
+    } else {
+        exactMVLut->size = 255;
+    }
+    exactMVLut->isSize = true;
+    auto typelist = type->to<IR::Type_SpecializedCanonical>()->arguments;
+    auto type0 =  typelist->at(0);
+    if (!type0->is<IR::Type_Bits>()) {
+        ::error("Invalid key type '%1%' for '%2%' 'Type_Bits' expected.", type0, d->externalName());
+        return;
+    }
+    auto type1 = typelist->at(1);
+    auto hwblk = new LookupHwBlocks();
+    hwblk->name = "MOD";
+    // hwblk->id = MOD;
+    hwblk->hw_interface = "CP_CHANNEL";
+    hwblk->hw_resource = resname;
+    hwblk->hw_resource_id = 7;
+    exactMVLut->matchAttributes = new LookupMatchAttributes();
+    exactMVLut->matchAttributes->hardware_blocks.push_back(hwblk);
+    UpdateMatchKeys(exactMVLut, type0);
+    UpdateImmediateFields(exactMVLut, type1);
+    contextLutTables.push_back(exactMVLut);
+}
+
+void DpdkContextGenerator::outputImmediateField(Util::JsonArray* immFields,
+        ImmediateFields* immfld) {
+    auto* immField = new Util::JsonObject();
+    immField->emplace("param_name", immfld->param_name);
+    immField->emplace("param_handle", immfld->param_handle);
+    immField->emplace("dest_start", immfld->dest_start);
+    immField->emplace("bitwidth", immfld->bitwidth);
+    immFields->append(immField);
+}
+
+void DpdkContextGenerator::outputKeys(Util::JsonObject* keyJsonObj,
+    std::vector<struct MatchKeyField*> &keylist, bool isLut) {
+    auto* keyJsons = new Util::JsonArray();
+    keyJsonObj->emplace("match_key_fields", keyJsons);
+
+    for (auto keyInfo : keylist) {
+        auto* keyJson = new Util::JsonObject();
+        keyJsons->append(keyJson);
+        keyJson->emplace("name", keyInfo->name);
+        if (keyInfo->isInstanceName) {
+            keyJson->emplace("instance_name", keyInfo->instanceName);
+        }
+        if (keyInfo->isFieldName) {
+            keyJson->emplace("field_name", keyInfo->fieldName);
+        }
+        if (keyInfo->matchType == exact) {
```

Comment:
keyJson->emplace("match_type", toStr(keyInfo->matchType)); 
 The above statement should give you the right match kind string instead of having these if-else-ifs

Reply:
can't be done, it's for Node* not for enum